src/ILLink.Tasks/LinkTask.cs

```diff
@@ -235,7 +235,7 @@ public class ILLink : ToolTask
 
 				var taskDirectory = Path.GetDirectoryName (Assembly.GetExecutingAssembly ().Location);
 				// The linker always runs on .NET Core, even when using desktop MSBuild to host ILLink.Tasks.
-				_illinkPath = Path.Combine (Path.GetDirectoryName (taskDirectory), "netcoreapp3.0", "illink.dll");
+				_illinkPath = Path.Combine (Path.GetDirectoryName (taskDirectory), "net5.0", "illink.dll");
```

Comment:
The proposed msbuild property for the used .NETCoreApp version can also flow into this file by supplying the constant in the project file.